docs/plookup.md

```diff
@@ -0,0 +1,310 @@
+correspond to a pair of adjacent elements in $s$. See the image below for better visualization.
+
+![](attachments/plookup_visualization.png)
+
+Below is the biconditional proof for our theorem.
+
+1. Suppose first that $f \subset t$ and $s \in \mathbb{F}^{n+d}$ is $(f, t)$ sorted by $t$.
+
+$s$ is sorted by $t$, so for each $j \in \lbrack d−1 \rbrack$, there is a distinct index $i \in \lbrack n+d−1 \rbrack$ such that $(
+t_j , t_{j+1}) = (s_i , s_{i+1}).$ The corresponding factors in $F, G$ are equal. That is,
+
+$$
+(\gamma + \frac{t_j + \beta t_{j+1}}{1 + \beta }) = (\gamma + \frac{s_i + \beta s_{i+1}}{1 + \beta })
+$$
+
+Let $P' \subset \lbrack n + d − 1 \rbrack$ the set of these $d − 1$ indices $i$ , and let $P := \lbrack n + d − 1 \rbrack \ \backslash
+\ P'$ . The $n$ indices $i \in P$ are such that $s_i = s_{i+1}$, and $\lbrace s_i\rbrace_{i\in P}$ equals $\lbrace f_i
+\rbrace_ {i\in \lbrack n \rbrack}$ as multisets. That is, we have a one-to-one map $j : P \rightarrow \lbrack n \rbrack$ such that for
+each $i
+\in P, s_i = f_{j(i)}$ . For each $i \in P$, the corresponding factor of $G$, will be
+
+$$
+\gamma + \frac{s_i + \beta s_{i+1}}{1 + \beta } = \gamma + s_i = \gamma + f_{j(i)},
+$$
+
+2. Assume $F \equiv G$ as polynomials in $\mathbb{F}\lbrack \beta , \gamma \rbrack$
+
+Then $F \equiv G$ also as elements of $\mathbb{F}(\beta )\lbrack \gamma \rbrack$. Since $\mathbb{F}(\beta )\lbrack \gamma \rbrack$ is a
+unique factorization domain (more details in [Appendix B](#appendix-B)), we know that the linear factors of $F, G$, as written above
+must be equal. Thus, for each $i \in \lbrack d−1 \rbrack, G$ must have a factor equal to $(\gamma + (t_i +\beta t_{i+1})/(1 +\beta )).$
+In other words, for some $j \in \lbrack n+d−1 \rbrack,$
+
+$$
+(\gamma + \frac{t_i + \beta t_{i+1}}{1 + \beta }) = (\gamma + \frac{s_j + \beta s_{j+1}}{1 + \beta })
+$$
+
+which implies
+$t_i + \beta t_{i+1} = s_j + \beta s_{j+1}$, and therefore $t_i = s_j , t_{i+1} = s_{j+1}$. Call $P' \subset \lbrack n + d − 1 \rbrack$
+the set of these $d − 1$ indices $j$. For any index $j \in \lbrack n + d − 1 \rbrack \ \backslash \ P'$ , there must be a factor
+“coming from $f$” in $F$ that equals the corresponding factor in $G$. More precisely, for such $j$ there exists $i \in \lbrack n
+\rbrack$ such that
+
+$$\gamma + f_i = \gamma + \frac{s_j + \beta s_{j+1}}{1 + \beta },$$
+
+or equivalently
+$$f_i + \beta f_i = s_j + \beta s_{j+1}$$
+which implies $f_i = s_j = s_{j+1}$
+
+Thus, we know that whenever consecutive values in $s$ are different, they are exactly equal to two consecutive values in $t$, and all
+values of $f$ are values of $t$.
+
+### Protocol
+
+We have the above theorem, the rest is to prove $F \equiv G$, using a “grand product argument” similar
+to [PLONK's Permutation Argument](#appendix-A). It will be convenient to assume $d = n+1$ (if $d \leq n$ pad $t$ with $n − d + 1$
```

Comment:
```suggestion
to [PLONK's Permutation Argument](#appendix-A). It will be convenient to assume $d = n+1$ (if $d \leq n$ pad $t$ with $n − d + 1$
```
```suggestion
to [PLONK's Permutation Argument](#appendix-A). It will be convenient to assume $d = n+1$ (if $d \leq n$ then pad $t$ with $n − d + 1$
```